### PR TITLE
[Infra] Add Publish Step

### PR DESCRIPTION
**Pull request by dominikandreasseitz**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Upload release to PyPI
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
```
Comment (Roland-djee) [inline .github/workflows/publish.yml:10]: I guess we use the 22.04 image in other actions. Might be best to harmonise to `latest`.

**Review by Roland-djee — APPROVED** [review]
Thanks @nmheim. LGTM.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (nmheim) [reply .github/workflows/publish.yml:10]: good point, will do!

**Review by madagra — APPROVED** [review]
Thanks, @nmheim, LGTM.

**Pull request merged**